src/test/ui/proc-macro/derive-helper-shadowing.rs

```diff
@@ -8,18 +9,15 @@ use test_macros::empty_attr as empty_helper;
 #[empty_helper] //~ ERROR `empty_helper` is ambiguous
 #[derive(Empty)]
 struct S {
-    // FIXME No ambiguity, attributes in non-macro positions are not resolved properly
-    #[empty_helper]
+    #[empty_helper] //~ ERROR `empty_helper` is ambiguous
     field: [u8; {
-        // FIXME No ambiguity, derive helpers are not put into scope for non-attributes
-        use empty_helper;
+        use empty_helper; //~ ERROR `empty_helper` is ambiguous
```

Comment:
If this were `use crate::empty_helper`, is the attribute on `struct U` still ambiguous?

Reply:
The attribute on `struct U` is still ambiguous because two different `empty_helper`s still exist in scope at the `struct U` point, and derive helpers always produce an error in case of ambiguity ([shadowing is not allowed](https://github.com/rust-lang/rust/pull/54086)).